controller/konnect/reconciler_generic.go

```diff
@@ -777,15 +795,21 @@ func handleKongConsumerRef[T constraints.SupportedKonnectEntityType, TEnt constr
 			return res, errStatus
 		}
 
-		return ctrl.Result{}, fmt.Errorf("can't get the referenced KongConsumer %s: %w", nn, err)
+		return ctrl.Result{}, ReferencedKongConsumerDoesNotExist{
+			Reference: nn,
+			Err:       err,
+		}
 	}
 
 	// If referenced KongConsumer is being deleted, return an error so that we
 	// can remove the entity from Konnect first.
 	if delTimestamp := consumer.GetDeletionTimestamp(); !delTimestamp.IsZero() {
-		return ctrl.Result{}, ReferencedKongConsumerIsBeingDeleted{
-			Reference: nn,
-		}
+		return ctrl.Result{
+				Requeue:      true,
+				RequeueAfter: time.Until(delTimestamp.Time),
+			}, ReferencedKongConsumerIsBeingDeleted{
+				Reference: nn,
+			}
```

Comment:
How about not changing the result here - it's kind of an antipattern to return a value and an error at once. We can decide on requeueing on the caller side.

Reply:
👍 d56733225fa4af9c780bc8d747bd01cb02adc769

PTAL.